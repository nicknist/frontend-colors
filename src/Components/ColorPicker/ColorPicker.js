import React, { useState } from 'react';
import { ChromePicker } from 'react-color';
import lockIcon from './lock.png';
import unlock from './unlock.jpeg';

const ColorPicker = ({ locked, color, handleChange, lock, number }) => {
  const [displayColorPicker, setDisplayColorPicker] = useState(false);

  const popOver = {
    position: 'absolute',
    zIndex: '2'
  };

  const cover = {
    position: 'fixed',
    top: '0px',
    right: '0px',
    bottom: '0px',
    left: '0px'
  };

  const background = {
    margin: '10px',
    padding: '5px',
    width: '100px',
    height: '200px',
    background: `${color}`,
    borderRadius: '1px'
  }

  const icon = {
    background: 'white',
    width: '20px',
    height: '20px'
  }

  return (
    <article>
      <div className="color-rectangle" style={background} onClick={() => setDisplayColorPicker(!displayColorPicker)}></div>
      <p>{color}</p>
      <p>Hex Code</p>
      { displayColorPicker ?
        <div className="color-rectangle" style={popOver}>
        <div className="no-hover" id={`square${number}`} style={cover} onClick={() => setDisplayColorPicker(false)}/>
        <ChromePicker className="chrome-picker" disableAlpha={true} color={color} onChangeComplete={(color) => handleChange(color.hex, number)}/>
        </div> : null }
      { locked ? <img style={icon} alt="lock button" className="lock" src={lockIcon} onClick={() => lock(color, number, locked)}/> : <img style={icon} alt="unlock button" className="unlock" src={unlock} onClick={() => lock(color, number, locked)}/> }
    </article>
  )
}

export default ColorPicker;
