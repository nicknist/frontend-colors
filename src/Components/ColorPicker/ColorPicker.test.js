import React from 'react';
import ColorPicker from './ColorPicker';
import { shallow } from 'enzyme';

describe('ColorPicker', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<ColorPicker />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show the colorPicker when the div color-rectangle is clicked', () => {
    const wrapper = shallow(<ColorPicker />);
    wrapper.find('.color-rectangle').simulate('click');
    wrapper.update();

    expect(wrapper).toMatchSnapshot();
  });

  it('should close the colorPicker when the no-hover div is clicked', () => {
    const wrapper = shallow(<ColorPicker />);
    wrapper.find('.color-rectangle').simulate('click');
    wrapper.update();

    wrapper.find('.no-hover').simulate('click');
    wrapper.update();

    expect(wrapper).toMatchSnapshot();
  });

  it('should run handleChange when the ChromePicker has hit onChangeComplete', () => {
    const mockHandleChange = jest.fn();
    const fakeEvent = () => {};
    const wrapper = shallow(<ColorPicker handleChange={mockHandleChange}/>);
    wrapper.find('.color-rectangle').simulate('click');
    wrapper.update();

    wrapper.find('.chrome-picker').prop('onChangeComplete')(fakeEvent);
    wrapper.update();

    expect(mockHandleChange).toHaveBeenCalled();
  });

  it('should match snapshot when locked is true', () => {
    const wrapper = shallow(<ColorPicker locked={true}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should run lock function when lock className is clicked', () => {
    const mockLock = jest.fn();
    const wrapper = shallow(<ColorPicker locked={true} lock={mockLock}/>);
    wrapper.find('.lock').simulate('click');

    expect(mockLock).toHaveBeenCalled();
  });

  it('should run lock function when unlock className is clicked', () => {
    const mockLock = jest.fn();
    const wrapper = shallow(<ColorPicker locked={false} lock={mockLock}/>);
    wrapper.find('.unlock').simulate('click');

    expect(mockLock).toHaveBeenCalled();
  })
})
