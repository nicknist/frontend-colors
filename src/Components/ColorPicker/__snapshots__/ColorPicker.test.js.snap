// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker should close the colorPicker when the no-hover div is clicked 1`] = `
<article>
  <div
    className="color-rectangle"
    onClick={[Function]}
    style={
      Object {
        "background": "undefined",
        "borderRadius": "1px",
        "height": "200px",
        "margin": "10px",
        "padding": "5px",
        "width": "100px",
      }
    }
  />
  <p />
  <p>
    Hex Code
  </p>
  <img
    alt="unlock button"
    className="unlock"
    onClick={[Function]}
    src="unlock.jpeg"
    style={
      Object {
        "background": "white",
        "height": "20px",
        "width": "20px",
      }
    }
  />
</article>
`;

exports[`ColorPicker should match snapshot when locked is true 1`] = `
<article>
  <div
    className="color-rectangle"
    onClick={[Function]}
    style={
      Object {
        "background": "undefined",
        "borderRadius": "1px",
        "height": "200px",
        "margin": "10px",
        "padding": "5px",
        "width": "100px",
      }
    }
  />
  <p />
  <p>
    Hex Code
  </p>
  <img
    alt="lock button"
    className="lock"
    onClick={[Function]}
    src="lock.png"
    style={
      Object {
        "background": "white",
        "height": "20px",
        "width": "20px",
      }
    }
  />
</article>
`;

exports[`ColorPicker should match the snapshot 1`] = `
<article>
  <div
    className="color-rectangle"
    onClick={[Function]}
    style={
      Object {
        "background": "undefined",
        "borderRadius": "1px",
        "height": "200px",
        "margin": "10px",
        "padding": "5px",
        "width": "100px",
      }
    }
  />
  <p />
  <p>
    Hex Code
  </p>
  <img
    alt="unlock button"
    className="unlock"
    onClick={[Function]}
    src="unlock.jpeg"
    style={
      Object {
        "background": "white",
        "height": "20px",
        "width": "20px",
      }
    }
  />
</article>
`;

exports[`ColorPicker should show the colorPicker when the div color-rectangle is clicked 1`] = `
<article>
  <div
    className="color-rectangle"
    onClick={[Function]}
    style={
      Object {
        "background": "undefined",
        "borderRadius": "1px",
        "height": "200px",
        "margin": "10px",
        "padding": "5px",
        "width": "100px",
      }
    }
  />
  <p />
  <p>
    Hex Code
  </p>
  <div
    className="color-rectangle"
    style={
      Object {
        "position": "absolute",
        "zIndex": "2",
      }
    }
  >
    <div
      className="no-hover"
      id="squareundefined"
      onClick={[Function]}
      style={
        Object {
          "bottom": "0px",
          "left": "0px",
          "position": "fixed",
          "right": "0px",
          "top": "0px",
        }
      }
    />
    <ColorPicker
      className="chrome-picker"
      color={
        Object {
          "a": 1,
          "h": 250,
          "l": 0.2,
          "s": 0.5,
        }
      }
      disableAlpha={true}
      onChangeComplete={[Function]}
      styles={Object {}}
      width={225}
    />
  </div>
  <img
    alt="unlock button"
    className="unlock"
    onClick={[Function]}
    src="unlock.jpeg"
    style={
      Object {
        "background": "white",
        "height": "20px",
        "width": "20px",
      }
    }
  />
</article>
`;
